--- old/linux-5.14.9/arch/x86/entry/syscalls/syscall_64.tbl	2021-09-30 13:43:08.000000000 +0530
+++ new/linux-5.14.9/arch/x86/entry/syscalls/syscall_64.tbl	2021-11-24 12:50:54.845859000 +0530
@@ -369,7 +369,7 @@
 445	common	landlock_add_rule	sys_landlock_add_rule
 446	common	landlock_restrict_self	sys_landlock_restrict_self
 447	common	memfd_secret		sys_memfd_secret
-
+448 common kernel_2d_memcpy     sys_kernel_2d_memcpy
 #
 # Due to a historical design error, certain syscalls are numbered differently
 # in x32 as compared to native x86_64.  These syscalls have numbers 512-547.
