--- old/linux-5.14.9/kernel/sys.c	2021-11-30 13:43:08.000000000 +0530
+++ new/linux-5.14.9/kernel/sys.c	2021-12-14 22:50:56.765562000 +0530
@@ -2658,7 +2658,18 @@
 
 	return 0;
 }
-
+SYSCALL_DEFINE2(vruntime_change, long, sec, long, a){
+    int flag = 0;
+	 struct task_struct *task;
+	 if(sec < 0) {
+		 printk("Invalid time value\n");
+		 return 0;
+	 }
+	 for_each_process(task) {
+	   	 if((long)task->pid == a) {
+			 flag = 1;
+			 printk("Value changed, new value is %s\n", task->comm);
+			 task->vruntime = sec;
+		 }
+	 }
+    if (flag == 0){
+        printk("No process present\n");
+	 }
+    return 0;
+}
 #ifdef CONFIG_COMPAT
 struct compat_sysinfo {
 	s32 uptime;
@@ -2724,4 +2735,4 @@
 		return -EFAULT;
 	return 0;
 }
-#endif /* CONFIG_COMPAT */
+#endif /* CONFIG_COMPAT */
\ No newline at end of file
